### PR TITLE
Update pandas_df_utils.py

### PR DESCRIPTION
**Pull request by yueguoguo**
Fixed the issues of the negative sampler

### Description
Fix the issue in the negative sampling function

### Related Issues
N/A

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [x] I have added tests covering my contributions.
- [x] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: reco_utils/dataset/pandas_df_utils.py [hunk 1]
```diff
@@ -293,10 +290,8 @@ def negative_feedback_sampler(
         If the samples exceed the number of total possible negative feedback samples, it will be reduced to the number
         of all the possible samples.
         seed (int): seed for the random state of the sampling function.
-
```
Comment (miguelgfierro) [inline reco_utils/dataset/pandas_df_utils.py:296]: minor thing here, if we remove these intermediate lines, the sphinx documentation doesn't parse the docstring correctly

**Review by miguelgfierro — APPROVED** [review]

**Review by anargyri — APPROVED** [review]
I think we can make negative sampling more efficient. Currently we create all user item pairs, which  is time and memory consuming i.e. O(n_users x n_items). Instead, we could sample indexes iteratively, which would be O(n_sample).

File: reco_utils/dataset/pandas_df_utils.py (hunk 1 above)
Comment (miguelgfierro) [reply reco_utils/dataset/pandas_df_utils.py:296]: hey @yueguoguo, since this is a small fix and you are on leave, I'll go ahead and fix it

**Comment by miguelgfierro** [comment]
``` I think we can make negative sampling more efficient. Currently we create all user item pairs, which is time and memory consuming i.e. O(n_users x n_items). Instead, we could sample indexes in O(n_sample) time e.g. using numpy sampling functions.

+ 1 to using numpy

**Pull request merged**